client_example/24_grouped_conv_activation/grouped_convnd_fwd_convscale_reduce/common.hpp

```diff
@@ -0,0 +1,825 @@
+    float scale_out = float(std::rand()) / float(RAND_MAX);
+
+    // We have NDHWGC/GKZYXC/NDHWGK (x, weight, y) in memory space.
+    // However, CK's API only accepts lengths and strides with order of GNCDHW/GKCZYX/GNKDHW.
+    // Hence, we need to adjust the order of strides.
+    const std::array<ck::index_t, NumDimSpatial + 3> input_lengths{G, N, C, Di, Hi, Wi};
+    const std::array<ck::index_t, NumDimSpatial + 3> input_strides{
+        C, Di * Hi * Wi * G * C, 1, Hi * Wi * G * C, Wi * G * C, G * C};
+    const std::array<ck::index_t, NumDimSpatial + 3> weights_lengths{G, K, C, Z, Y, X};
+    const std::array<ck::index_t, NumDimSpatial + 3> weights_strides{
+        K * Z * Y * X * C, Z * Y * X * C, 1, Y * X * C, X * C, C};
+    const std::array<ck::index_t, NumDimSpatial + 3> output_lengths{G, N, K, Do, Ho, Wo};
+    const std::array<ck::index_t, NumDimSpatial + 3> output_strides{
+        K, Do * Ho * Wo * G * K, 1, Ho * Wo * G * K, Wo * G * K, G * K};
+
+    /*
+     * FP8 Convolution with Scaling
+     */
+    std::cout << "\n\nConvolution with scale Benchmarking:" << std::endl;
+    auto elementwise_op = ConvElementOp{ew::Scale{scale_in}, ew::Scale{scale_wei}, {}};
+    auto conv_ok        = ConvolutionScale<InDataType,
+                                    WeiDataType,
+                                    ConvOutDataType,
+                                    ConvElementOp,
+                                    InLayout,
+                                    WeiLayout,
+                                    OutLayout,
+                                    NumDimSpatial>(in,
+                                                   wei,
+                                                   conv_out,
+                                                   elementwise_op,
+                                                   input_lengths,
+                                                   input_strides,
+                                                   weights_lengths,
+                                                   weights_strides,
+                                                   output_lengths,
+                                                   output_strides);
+
+    if(!conv_ok)
+        return false;
+
+    /*
+     *  Scale with output weight and convert to FP8
+     */
+    std::cout << "\n\nElement-wise scale + convert Benchmarking:" << std::endl;
+    TensorScaleConvert<ConvOutDataType, OutDataType, NumDimSpatial>(
+        conv_out, out, scale_out, output_lengths, output_strides);
+
+    /*
+     *  Compute AMAX
+     */
+    std::cout << "\n\nAMAX Benchmarking:" << std::endl;
+    SimpleDeviceMem amax_device(sizeof(ConvOutDataType));
+    auto amax_host =
+        TensorFullReduction<ConvOutDataType,
+                            ConvOutDataType,
+                            ck::ReduceTensorOp::AMAX,
+                            NumDimSpatial>(conv_out, amax_device, output_lengths, output_strides);
+    return true;
```

Comment:
Looks like we treat 3 components' output in different ways: convolution returns bool ok/not ok, scale convert returns void and reduction returns scale value. Would it make sense to unify them to return either bool, void or value?

Reply:
Great suggestion! I will return status for each kernel.